packages/compass-connect/src/stores/index.js

```diff
@@ -540,8 +555,9 @@ const Store = Reflux.createStore({
    * Selects a saved connection.
    *
    * @param {Connection} connection - The connection to select.
+   * @param {String} connectionEntry - The connection form entry.
    */
-  onConnectionSelected(connection) {
+  onConnectionSelected(connection, connectionEntry) {
```

Comment:
What do you think about this: can we just avoid this whole connection entry business?

In the connection model we already have `isFavorite` that would give us a distinction between a favorite and the other 2 kind. To tell a recent from a new connection you can use `connection.lastUsed` instead

Reply:
from the conversation we had [here](https://docs.google.com/spreadsheets/d/1Wi7np8CooM0b3xPHZNd-iYst_DnKEo7FBswbnSXKoys/edit#gid=953290655), `is_recent` refers to if user clicked on the item in recents section of the connection list. 

However if we use `connection.lastUsed`, what determines if the connection is recent? How long before when it's not recent?

Reply:
no `connection.lastUsed` should be `null` in that case

Reply:
that actually works, but then with this approach things are more explicit. 

a connection can be favourite and still have `connection.lastUsed` as null, so we have to compute `is_recent` from two different properties - `connectionModel.isFavorite` and `connection.lastUsed`. 

similarly, if we introduce `is_new`, we will also have to compute it as well.

Reply:
In my opinion if the properties in the `connectionModel` are good enough I would really avoid to create new derived state that has to be kept in sync, that's usually worse and can introduce issues.

This looks quite straightforward, especially compared with dispatching new actions:

```
{
   is_new: !connection.lastUsed,
   is_recent: connection.lastUsed && !connection.isFavorite,
   is_favorite: connection.isFavorite
}
```